independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/RestClientRequestContext.java

```diff
@@ -115,6 +117,17 @@ public void abort() {
         restart(abortHandlerChain);
     }
 
+    @Override
+    protected Throwable unwrapException(Throwable t) {
+        var res = super.unwrapException(t);
+        if (res instanceof WebApplicationException) {
+            WebApplicationException webApplicationException = (WebApplicationException) res;
+            return new ClientWebApplicationException(webApplicationException.getMessage(), webApplicationException,
```

Comment:
The `WebApplicationException` cause should be propagated here

Reply:
I think it should.

I'm only wondering if it wouldn't be slightly better if `webApplication.getCause()` was passed as the cause and not `webApplication`

Reply:
I'm not sure, it does loose a bit of context as to where the original exception came from (e.g. a filter could directly throw it with no cause.)

Reply:
true, let's leave it